### PR TITLE
'Revision' make target

### PR DESCRIPTION
**Pull request by jeremywmoore**
1. Adds a new `revision` target that will initialize a new db migration file.
2. Adds help comments to the `revision` and `migrate` targets so they are listed when running `make help`

### CODE REVIEW

**Comment by jeremywmoore** [comment]
```
% make
fix         Format the code
help        Print the help message
lint        Lint the code
migrate     Apply migrations
revision    Create a new migration
run         Run the app
test        Run the test suite
```

when running without specifying a message name:
```
% make revision
Makefile:31: *** 'message' must be set when invoking the revision target, eg `make revision message="short message"`.  Stop.
```

successful run:
```
% make revision message="short message"
. ./dev_env.sh && \
	poetry run flask db revision -m "short message"
  Generating /Users/jeremymoore/projects/hushline/migrations/versions/82c7cc8d5b80_short_message.py ...  done
```

**Review by rmlibre — CHANGES_REQUESTED** [review]
Are there places in the docs where this should be added? I see this one:

https://github.com/scidsg/hushline/blob/95c6c5c76ab01212bb30e0220f5d3d11e6a60f1d/docs/DEV.md?plain=1#L51-L57

**Review by rmlibre — COMMENTED** [review]
#### PS: 

I ran this & the migration commands I used for the last PR [#377], in the current [#411], but both failed to pick up any changes.

Do new tables need to be registered somewhere specific in order to be detected & integrated with the migrations?

File: docs/DEV.md [hunk 1]
```diff
@@ -26,13 +26,6 @@ You also need Rust to install some of the Python dependencies. Install [rustup](
 curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
 ```
 
-Using pipx, install Python packages required for linting and static type checking:
```
Comment (jeremywmoore) [inline docs/DEV.md:29]: these tools are installed by poetry

**Comment by jeremywmoore** [comment]
Added the new target to the docs and then cleaned up the targets so users don't have to wrap the calls with `poetry run`.

**Comment by jeremywmoore** [comment]
> #### PS:
> I ran this & the migration commands I used for the last PR [#377], in the current [#411], but both failed to pick up any changes.
> 
> Do new tables need to be registered somewhere specific in order to be detected & integrated with the migrations?

After creating the new revision file you should populate it with alembic operations to create the table and columns.

**Review by rmlibre — APPROVED** [review]
👍 I like the shortening of the commands

**Review by micahflee — APPROVED** [review]

**Pull request merged**